File: src/main/java/com/example/ProjectManagement/Controller/ProjectManagementController.java
package com.example.ProjectManagement.Controller;


import com.example.ProjectManagement.DTO.ProjectDto;
import com.example.ProjectManagement.Service.ProjectManagementService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/projects")
public class ProjectManagementController {

    private final ProjectManagementService projectManagementService;

    public ProjectManagementController(ProjectManagementService projectManagementService) {
        this.projectManagementService = projectManagementService;
    }

    @PostMapping
    public String createProject(@RequestBody ProjectDto projectDto) {
        return projectManagementService.createProject(projectDto);
    }

    @GetMapping("/{id}")
    public ProjectDto getProject(@PathVariable String id) {
        return projectManagementService.getProjectById(id);
    }

    @PutMapping("/{id}")
    public String updateProject(@PathVariable String id, @RequestBody ProjectDto projectDto) {
        return projectManagementService.updateProject(id, projectDto);
    }

    @DeleteMapping("/{id}")
    public String deleteProject(@PathVariable String id) {
        return projectManagementService.deleteProject(id);
    }

    @GetMapping // Supports GET
    public ResponseEntity<?> getProjects() {
        // Handle the request
        return ResponseEntity.ok().body("Projects Retrieved");
    }
}

File: src/main/kotlin/com/charitan/management/project/ProjectManagementController.kt
package com.charitan.management.project

import com.charitan.management.project.dto.ProjectHaltReasonDto
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RequestMapping("/api/admin/project")
@RestController
class ProjectManagementController(
    private val projectManagementService: ProjectManagementExternalService,
) {
    @PostMapping("/halt/{id}")
    suspend fun haltProject(
        @PathVariable id: String,
        @RequestBody dto: ProjectHaltReasonDto,
    ) {
        projectManagementService.halt(id, dto)
    }

    @PostMapping("/approve/{id}")
    suspend fun approveProject(
        @PathVariable id: String,
    ) {
        projectManagementService.approve(id)
    }
}
